### PR TITLE
fix: drop .group from incoming messages & exclude it from visible check

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/receiver/contentMessage.ts [hunk 1]
```diff
@@ -317,7 +319,7 @@ function shouldDropBlockedUserMessage(content: SignalService.Content): boolean {
   if (!isMessageDataMessageOnly) {
     return true;
   }
-  const data = content.dataMessage;
+  const data = content.dataMessage as SignalService.DataMessage; // forcing it as we do know this field is set based on last line
```
Comment (yougotwill) [inline ts/receiver/contentMessage.ts:322]: Do you mean because we use isMessageDataMessageOnly?

File: ts/receiver/contentMessage.ts [hunk 2]
```diff
@@ -343,11 +345,21 @@ export async function innerHandleSwarmContentMessage(
     const content = SignalService.Content.decode(new Uint8Array(plaintext));
     perfEnd(`SignalService.Content.decode-${envelope.id}`, 'SignalService.Content.decode');
 
-    const blocked = await isBlocked(envelope.source);
+    /**
+     * senderIdentity is set ONLY if that message is a closed group message.
+     * If the current message is a closed group message,
+     * envelope.source is going to be the real sender of that message.
+     *
+     * When receiving a message from a user which we blocked, we need to make let
```
Comment (yougotwill) [inline ts/receiver/contentMessage.ts:353]: * we need to let the control message through

File: ts/receiver/contentMessage.ts [hunk 3]
```diff
@@ -343,11 +345,21 @@ export async function innerHandleSwarmContentMessage(
     const content = SignalService.Content.decode(new Uint8Array(plaintext));
     perfEnd(`SignalService.Content.decode-${envelope.id}`, 'SignalService.Content.decode');
 
-    const blocked = await isBlocked(envelope.source);
+    /**
+     * senderIdentity is set ONLY if that message is a closed group message.
+     * If the current message is a closed group message,
+     * envelope.source is going to be the real sender of that message.
```
Comment (yougotwill) [inline ts/receiver/contentMessage.ts:351]: So in a closed group the senderIdentity is the admin or member and the source is the closed group key? I'm a bit confused?

File: ts/receiver/dataMessage.ts [hunk 4]
```diff
@@ -37,14 +37,13 @@ function cleanAttachment(attachment: any) {
...
-  const { quote, group } = decrypted;
+  const { quote } = decrypted;
 
   // Here we go from binary to string/base64 in all AttachmentPointer digest/key fields
 
-  // we do not care about group.avatar on Session
-  if (group && group.avatar !== null) {
-    group.avatar = null;
-  }
+  // we do not care about group on Session
```
Comment (yougotwill) [inline ts/receiver/dataMessage.ts:44]: maybe we don't care on desktop but Android still uses it?

File: ts/receiver/dataMessage.ts [hunk 5]
```diff
@@ -80,33 +79,17 @@ function cleanAttachments(decrypted: SignalService.DataMessage) {
...
 
   return (
-    !flags &&
-    isEmpty(body) &&
-    isEmpty(attachments) &&
-    isEmpty(group) &&
-    isEmpty(quote) &&
-    isEmpty(preview) &&
-    isEmpty(openGroupInvitation)
+    !!flags ||
```
Comment (yougotwill) [inline ts/receiver/dataMessage.ts:86]: Nice! 😂

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/receiver/dataMessage.ts (hunk 4 above)
Comment (Bilb) [reply ts/receiver/dataMessage.ts:44]: updated with Session Desktop

File: ts/receiver/contentMessage.ts (hunk 2 above)
Comment (Bilb) [reply ts/receiver/contentMessage.ts:353]: updated

File: ts/receiver/contentMessage.ts (hunk 1 above)
Comment (Bilb) [reply ts/receiver/contentMessage.ts:322]: yeah that's what I meant but it doesn't actually check for undefined.
Updated with an early return to true (message dropped) if the message has no dataMessage at all.
(if there is no data message, it cannot be a control message)

File: ts/receiver/contentMessage.ts (hunk 3 above)
Comment (Bilb) [reply ts/receiver/contentMessage.ts:351]: updated comment

File: ts/receiver/contentMessage.ts (hunk 1 above)
Comment (yougotwill) [reply ts/receiver/contentMessage.ts:322]: Cool

File: ts/receiver/contentMessage.ts (hunk 3 above)
Comment (yougotwill) [reply ts/receiver/contentMessage.ts:351]: 👍

File: ts/receiver/contentMessage.ts (hunk 2 above)
Comment (yougotwill) [reply ts/receiver/contentMessage.ts:353]: 👍

**Review by yougotwill — APPROVED** [review]
LGTM 🚀

**Pull request merged**